mtags/src/main/scala-3/scala/meta/internal/mtags/MtagsEnrichments.scala

```diff
@@ -144,6 +144,18 @@ object MtagsEnrichments extends CommonMtagsEnrichments:
       upd.rawParamss = paramsWithFlags
       upd
     end withUpdatedTpe
+
+    // Returns true if this symbol is locally defined from an old version of the source file.
+    def isStale: Boolean =
+      sym.sourcePos.span.exists && {
+        val source = ctx.source
+        if source ne sym.source then
+          !source.content.startsWith(
+            sym.decodedName.toString(),
+            sym.sourcePos.span.point,
+          )
```

Comment:
This line looks suspicious :thinking: 
Wouldn't it be enough to check only `source ne ctx.source` ?

Reply:
This is copied from Scala 2 implementation, maybe we should copy over the comment? Or maybe this is no longer neccessary in Scala 3?

Reply:
> Wouldn't it be enough to check only source ne ctx.source ?

Thanks, its enough, at least for match completions tests

Reply:
hmm. I missed that it's a different `startsWith(.., offset)`. 
Then it makes sense for me, feel free to return it back if it's needed

I thought that it was usual `startsWith` at idx 0 and didn't get what is the purpose of this clause :facepalm:

Reply:
Turns out it actually is needed, otherwise it marks `not-stale` symbols as `stale` so I returned it.

Reply:
Yep - I saw failed tests. Sorry for non-relevant suggestion